substrate/frame/examples/authorization-tx-extension/src/extensions.rs

```diff
@@ -0,0 +1,132 @@
+/// data that follows this extension in the `TransactionExtension` pipeline, their implications and
+/// the call. Essentially re-sign the transaction from this point onwards in the pipeline by using
+/// the `inherited_implication`, as shown below.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct AuthorizeCoownership<T, Signer, Signature> {
+	inner: Option<AuthCredentials<Signer, Signature>>,
+	_phantom: PhantomData<T>,
+}
+
+impl<T: Config, Signer, Signature> Default for AuthorizeCoownership<T, Signer, Signature> {
+	fn default() -> Self {
+		Self { inner: None, _phantom: Default::default() }
+	}
+}
+
+impl<T: Config, Signer, Signature> AuthorizeCoownership<T, Signer, Signature> {
+	/// Creates an active extension that will try to authorize the coownership origin.
+	pub fn new(first: (Signer, Signature), second: (Signer, Signature)) -> Self {
+		Self { inner: Some(AuthCredentials { first, second }), _phantom: Default::default() }
+	}
+}
+
+impl<T: Config, Signer, Signature> fmt::Debug for AuthorizeCoownership<T, Signer, Signature> {
+	#[cfg(feature = "std")]
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "AuthorizeCoownership")
+	}
+
+	#[cfg(not(feature = "std"))]
+	fn fmt(&self, _: &mut fmt::Formatter) -> fmt::Result {
+		Ok(())
+	}
+}
+
+impl<T: Config + Send + Sync, Signer, Signature> TransactionExtension<T::RuntimeCall>
+	for AuthorizeCoownership<T, Signer, Signature>
+where
+	Signer: IdentifyAccount<AccountId = T::AccountId> + Parameter + Send + Sync + 'static,
+	Signature: Verify<Signer = Signer> + Parameter + Send + Sync + 'static,
+{
+	const IDENTIFIER: &'static str = "AuthorizeCoownership";
+	type Implicit = ();
+	type Val = ();
+	type Pre = ();
+
+	fn validate(
+		&self,
+		mut origin: DispatchOriginOf<T::RuntimeCall>,
+		_call: &T::RuntimeCall,
+		_info: &DispatchInfoOf<T::RuntimeCall>,
+		_len: usize,
+		_self_implicit: Self::Implicit,
+		inherited_implication: &impl codec::Encode,
+	) -> ValidateResult<Self::Val, T::RuntimeCall> {
+		// If the extension is inactive, just move on in the pipeline.
+		let Some(auth) = &self.inner else {
+			return Ok((ValidTransaction::default(), (), origin));
+		};
+		let first_account = auth.first.0.clone().into_account();
+		let second_account = auth.second.0.clone().into_account();
+
+		// Construct the payload to sign using the `inherited_implication`.
+		let msg = inherited_implication.using_encoded(sp_io::hashing::blake2_256);
+
+		// Both parties' signatures must be correct for the origin to be authorized.
+		// In a prod environment, we're just return a `InvalidTransaction::BadProof` if the
+		// signature isn't valid, but we return these custom errors to be able to assert them in
+		// tests.
+		if !auth.first.1.verify(&msg[..], &first_account) {
+			Err(InvalidTransaction::Custom(100))?
+		}
+		if !auth.second.1.verify(&msg[..], &second_account) {
```

Comment:
This is like a fully state-less, offchain, 2/2 multisig

Reply:
Yeah, the pallet functionality isn't really useful, this example just demonstrates how to use the origin mutation and the `inherited_implication. Hopefully this example shows how `TransactionExtension` can be used for some actually useful signature authorization schemes.